networks/rpc/http_datadog.go

```diff
@@ -0,0 +1,145 @@
+package rpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	httptrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type DatadogTracer struct {
+	Tags           []string
+	Service        string
+	KlaytnResponse bool
+}
+
+func newDatadogTracer() *DatadogTracer {
+	ddTraceEnabled := os.Getenv("DD_TRACE_ENABLED")
+	if strings.ToLower(ddTraceEnabled) != "true" {
+		return nil
+	}
+
+	tracer.Start()
+
+	tags := strings.Split(os.Getenv("DD_TRACE_HEADER_TAGS"), ",")
+	service := os.Getenv("DD_SERVICE")
+	klaytnResponse := strings.ToLower(os.Getenv("DD_KLAYTN_RPC_RESPONSE")) == "true"
+
+	return &DatadogTracer{tags, service, klaytnResponse}
+}
+
+func newDatadogHTTPHandler(ddTracer *DatadogTracer, handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				logger.ErrorWithStack("Datadog http handler panic", "err", err)
+			}
+		}()
+
+		reqMethod := ""
+		reqParam := ""
+
+		// parse RPC requests
+		reqs, isBatch, err := getRPCRequests(r)
+
+		if err != nil || len(reqs) < 1 {
+			// The error will be handled in `handler.ServeHTTP()` and printed with `printRPCErrorLog()`
+			logger.Debug("failed to parse RPC request", "err", err, "len(reqs)", len(reqs))
+		} else {
+			reqMethod = reqs[0].method
+			if isBatch {
+				reqMethod += "_batch"
+			}
+			encoded, _ := json.Marshal(reqs[0].params)
+			reqParam = string(encoded)
+		}
+
+		// datadog transaction name contains the first API method of the request
+		resource := r.Method + " " + r.URL.String() + " " + reqMethod
```

Comment:
`fmt.Sprintf` looks better here.
```suggestion
		resource := fmt.Sprintf("%s %s %s", r.Method, r.URL.String(), reqMethod)
```

Reply:
Thanks. changed.